### PR TITLE
support evaluate in orca tf estimator

### PR DESCRIPTION
**Pull request by yangw1234**
Usage:
```python
    data_shard = data_shard.transform_shard(transform)

     est = Estimator.from_graph(
         inputs=[model.user, model.item],
         labels=[model.label],
         loss=model.loss,
         optimizer=tf.train.AdamOptimizer(),
         metrics={"loss": model.loss})
     result = est.evaluate(data_shard)
     assert "loss" in result
     print(result)
```

Fix https://github.com/analytics-zoo/orca/issues/51

### CODE REVIEW

**Comment by yangw1234** [comment]
jenkins passed http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/3560/

**Comment by jenniew** [comment]
Does the evaluation do the shuffling to change the order?

**Comment by jenniew** [comment]
> Does the evaluation do the shuffling to change the order?

Seems no shuffling. If so, we can use in NCF case.

File: pyzoo/zoo/orca/learn/tf/estimator.py [hunk 1]
```diff
@@ -154,6 +170,7 @@ def __init__(self, *, inputs, outputs, labels, loss,
         else:
             self.sess = sess
         self.model_dir = model_dir
+        self.real_batch_size = tf.shape(self.inputs[0])[0]
```
[the remark this answers is not in the document]
Comment (yangw1234) [reply pyzoo/zoo/orca/learn/tf/estimator.py:173]: users do not need to care about that. When averaging metrics, we need to take into account the real_batch_size when averaging.

**Comment by yangw1234** [comment]
No, the evaluation does not change data placement.

**Comment by jenniew** [comment]
LGTM

File: pyzoo/zoo/orca/learn/tf/estimator.py [hunk 2]
```diff
@@ -204,16 +210,22 @@ def predict(self, data, batch_size=32):
...
+        dataset = _to_dataset(data, batch_size=-1, batch_per_thread=batch_size)
 
         flat_inputs = nest.flatten(self.inputs)
         flat_outputs = nest.flatten(self.outputs)
         tfnet = TFNet.from_session(sess=self.sess, inputs=flat_inputs, outputs=flat_outputs)
         return tfnet.predict(dataset)
+
+    def evaluate(self, data, batch_size=32):
+        assert self.metrics is not None, \
+            "output is None, it should not be None in prediction"
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/tf/estimator.py:222]: metrics is None

Comment (yangw1234) [reply pyzoo/zoo/orca/learn/tf/estimator.py:222]: will open an pr to fix this
